test/GPv2Settlement/Settle/Settle.t.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.8.26;
+
+import {Vm} from "forge-std/Test.sol";
+
+import {Harness, Helper} from "../Helper.sol";
+
+import {GPv2Settlement, GPv2Interaction} from "src/contracts/GPv2Settlement.sol";
+
+import {Settlement} from "test/libraries/Settlement.sol";
+import {SettlementEncoder} from "test/libraries/encoders/SettlementEncoder.sol";
+
+// solhint-disable func-name-mixedcase
+contract Settle is Helper {
+    using Settlement for Harness;
+    using SettlementEncoder for SettlementEncoder.State;
+
+    function setUp() public override {
+        super.setUp();
+    }
+
+    function test_allowlist_rejects_transactions_from_non_solvers() public {
+        vm.expectRevert("GPv2: not a solver");
+        settlement.settle(encoder.encode(settlement));
```

Comment:
What do you think about merging the `Settlement` library into the `Harness` in this PR? This would reduce the number of "logical" places where to look for this `settle` function to two. Not sure if it's needed somewhere else, but in case I might still be in favor of extending `Harness`, keeping the library, but not using `using` here.

Reply:
Generally extending from the Helper I think is fine. Done.